src/main/java/com/iota/iri/conf/converters/InetAddressConverter.java

```diff
@@ -0,0 +1,28 @@
+package com.iota.iri.conf.converters;
+
+import com.beust.jcommander.IStringConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * Used by JCommander to convert a string from command line into an InetAddress.
+ * Refer to {@link com.iota.iri.conf.APIConfig.Descriptions#REMOTE_TRUSTED_API_HOSTS} for more details.
+ *
+ * @author legacycode
+ */
+public class InetAddressConverter implements IStringConverter<InetAddress> {
+    private static final Logger log = LoggerFactory.getLogger(InetAddressConverter.class);
+
+    @Override
+    public InetAddress convert(String address) {
+        try {
+            return InetAddress.getByName(address.trim());
+           } catch (UnknownHostException e) {
+            log.error("Invalid value ({}) for paramater: {}", address.trim(), e.getMessage());
+        }
+        return null;
+    }
+}
```

Comment:
Hmm,

When you use JCommander without setters you will need this class.
Why not just put `InetAddress.getByName(address.trim())` in the setter?
This is just more boiler plate code... So please remove :-)

You can also rethrow the `UnknowHostException` as a `ParameterException`.
If the user enters a bad value it was probably by mistake and it is best for IRI not to start to give the user the opportunity to correct his mistake.

By the way the reason we are using setter is because in the future I think that most configs should have some validation/conversion and it nicer to do in the setter then in an external class. Of course this is my opinion and you can disagree with me on that...

Reply:
> In my opinion BaseIotaConfig.java should only be an entity class and validation and conversion logic should be moved into JCommander converters and validators.

I don't feel like it is such a pain. It is fine to add logic in setters. You can also gain re-usability with private methods.
You can claim that the `BaseIotaConfig` class is getting very big so it may be better to do it your way (and you have a good point). I don't see it as much of a priority to be honest or a big issue.

>  also would like to drop/move the support of Jackson configuration file and migrate to JCommander configuration files provided by the @syntax

I agree that if we can drop a dependency then it is a good idea